pretrain/scripts/v3-high-quality-cpt/megatron_data_formatter.py

```diff
@@ -0,0 +1,161 @@
+"""
+
+import csv
+from argparse import ArgumentParser
+from pathlib import Path
+
+import yaml
+
+# Argument parser setup
+argparser = ArgumentParser()
+argparser.add_argument("yaml_config")
+argparser.add_argument(
+    "-f",
+    "--force",
+    action="store_true",
+    default=False,
+    help="Ignore gnore duplicate dataset entries",
+)
+argparser.add_argument(
+    "-d",
+    "--display",
+    action="store_true",
+    default=False,
+    help="Display detailed dataset statistics including token sizes and paths",
+)
+
+
+def process_info(
+    info_file: Path, repeat: float, filters: list[str] | None = None
+) -> list[str]:
+    """
+    Processes dataset info from the CSV file, applies the repeat factor to the token count,
+    and optionally filters based on the provided patterns.
+
+    Args:
+        info_file (Path): Path to the CSV file containing dataset information.
+        repeat (float): Multiplier for token sizes.
+        include_chars (list[str], optional): List of filter patterns. Defaults to an empty list.
+
+    Returns:
+        list[str]: List of formatted dataset strings (token size and data path).
+    """
+    data_paths = []
+    with open(info_file, newline="", encoding="utf-8") as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            token_count = int(int(row[2]) * repeat + 0.5)
+            data_path = row[1]
+            # Filter by filters if provided, otherwise add the path
+            if filters is None or any(pattern in data_path for pattern in filters):
+                data_paths.append(f"{token_count} {data_path}")
+    return data_paths
+
+
+def check_load_dataset(
+    train_data_path: str, force: bool = False, display_details: bool = False
+) -> None:
+    """
+    Checks and displays the dataset details including token sizes and file names.
+    Optionally detects and handles duplicate entries.
+
+    Args:
+        train_data_path (str): The TRAIN_DATA_PATH string containing dataset paths and token sizes.
+        force (bool, optional): If True, ignore duplicates and continue. Defaults to False.
+        display_details (bool, optional): If True, display detailed token size breakdown. Defaults to False.
+    """
+    total_token_size = 0
+    file_checker = set()
+
+    if display_details:
+        lang_total_tokens: dict[str, int] = {}
+        print(f"{'Lang':<5} {'File Name':<20} {'Token Size':>15}")
+
+    data_entries = train_data_path.strip().split()
+    for i in range(0, len(data_entries), 2):
+        token_size = int(data_entries[i])
+        data_path = Path(data_entries[i + 1])
+        lang = data_path.parent.name
+        file_name = data_path.stem
+        combination = f"{lang}/{file_name}"
+
+        if combination in file_checker:
+            common_msg = f"Duplicate entry for {combination}."
+            if not force:
+                raise ValueError(f"{common_msg} Exiting.")
+            else:
+                print(f"Warning: {common_msg}")
```

Comment:
または明示的に警告を出したいなら `warnings.warn`
```suggestion
                logging.warn(common_msg)
```